sys/uri_parser/uri_parser.c

```diff
@@ -103,22 +110,46 @@ bool _consume_port(uri_parser_result_t *result, const char *ipv6_end,
 
     /* check if match */
     if (port_begin && (port_begin[0] == ':')) {
-        /* port should be at least one character, => + 1 */
-        if (port_begin + 1 == authority_end) {
+        port_begin++;
+
+        /* port should be at least one character long */
+        if (port_begin == authority_end) {
+            return false;
+        }
+
+        port_str_len = authority_end - port_begin;
+
+        /* Verify that the port number is up to 5 (random) chars in size */
+        if (port_str_len > MAX_PORT_STR_LEN) {
+            return false;
+        }
+
+        /* Verify that all characters of the port are numerical */
+        for (int i = 0; i < port_str_len; ++i) {
+            if (!((port_begin[i] >= '0') && (port_begin[i] <= '9'))) {
+                return false;
+            }
+        }
+
+        /* Verify that the next character, after the port, is an invalid
```

Comment:
there's `scn_u32_dec()` in the fmt module that works like  `atol()`, but with character limit.
It could be used instead this check and atol.

(If we're fine with "accepting" `host:123ba/" as port=123, you with scn_u32_dec() you can also skip the verification starting in line  128).

Reply:
I don't feel like it would be an enhancement but rather an equal exchange (code reuse < - > more dependencies) and I wouldn't remove the character check. Failing with an illegal URI should be encouraged behaviour instead of "somewhat (un)defined" behaviour. Compare it to `curl`:
```
curl http://google.com:80abcd/
curl: (3) URL using bad/illegal format or missing URL
```